### DISCUSSION TITLE
Please Bring Back Web Support

### DISCUSSION THREAD

**Discussion by bnussman** [react-native-maps/react-native-maps#4741] [Feature requests]
## Description 🗒️ 

- Please bring back web support
- I can't be the only one stuck on v0.31.1 to retain the ability to develop my app in the browser. It's frustrating.
- I'm not asking for a map to render in the browser, **I'm only asking for the ability for my app to still render like the behavior in 0.31.1**
- Bringing this back is important because it enabled developers to develop their app in the browser without a phsycal device or simulator
- Using Expo SDK 48 and the Metro bundler

## Before (0.31.1) 🎊 

> **Note**:  Notice how the app actually renders. The red box is where a map shows up on iOS and android

![Screenshot from 2023-05-09 23-04-46](https://github.com/react-native-maps/react-native-maps/assets/6440455/3d8d80d8-d0d8-4633-9cf3-f250e65cac4a)

## After (1.7.1) 😠 

```sh
Unable to resolve "../Utilities/Platform" from "../node_modules/react-native/Libraries/ReactPrivate/ReactNativePrivateInterface.js"
```

Metro Error
![Screenshot from 2023-05-09 22-57-55](https://github.com/react-native-maps/react-native-maps/assets/6440455/d281597b-ac0d-46ad-bf53-5d287af054f3)

Webapck Error
![Screenshot from 2023-05-09 23-13-44](https://github.com/react-native-maps/react-native-maps/assets/6440455/b4a3f531-be04-4372-ae86-11f0dab1089e)

### Reproducible sample code

```TSX
import { MapView } from 'react-native-maps';

export default function App() {
  return (
    <MapView
      initialRegion={{
        latitude: 37.78825,
          longitude: -122.4324,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421,
      }}
    />
  );
}
```

### Steps to reproduce

1. Create a new Expo project with `npx create-expo-app map-issue`
2. `cd map-issue`
3. Install web things that expo make you install to use web with `npx expo install react-native-web@~0.18.10 react-dom@18.2.0 @expo/webpack-config@^18.0.1`
4. Put a `<MapView />` in  the App.js
5. `npm run web`
6. The app won't load in the web

### Expected result

I'd expect my app to still run successfully in the web like it does with `"react-native-maps": "0.31.1"`, even if it renders a placeholder instead of an actual map.

I'd expect the functionality of the latest `react-native-maps` to have equal or better web compatibility than `"react-native-maps": "0.31.1"`, but it doesn't even support web at all anymore 

### Actual result

Refer to errors in the _summary_ section. The app errors when using expo's web feature with **both** Expo's webpack and metro web bundlers.

For context, You can opt-in to metro for web by specifying `bundler: "metro"` in your expo config.

```typescript
export default {
  name: "Beep",
  displayName: "Beep",
  expo: {
    platforms: ["ios", "android", "web"],
    web: {
      favicon: "./assets/favicon.png",
      bundler: "metro",
    },
  },
};

```

### React Native Maps Version

Web support broken in both 1.7.1 and 2.0.0-beta.11. Last known working is 0.31.1

### What platforms are you seeing the problem on?

Web

### React Native Version

0.71.3

### What version of Expo are you using?

SDK 48

### Device(s)

The browser 😐

**Comment by bnussman** [comment]
Duplicate of https://github.com/react-native-maps/react-native-maps/issues/4641, but can we at least have a discussion on this insted of just closing the issue? 

I'd be willing to contribute if I had some instruction on some possible paths forward

**Comment by YoussefHenna** [comment]
You can use this [patch ](https://github.com/draftbit/react-native-jigsaw/blob/dd350845cbbfa3252d93527b01a2b25e3e556aef/patches/react-native-maps%2B1.3.2.patch#LL1) to get it working for now. Might need to be recreated for the latest version, just follow the steps [here](https://github.com/react-native-maps/react-native-maps/issues/4641#issuecomment-1473775067) manually in `node_modules` then create a patch

**Reply by bnussman** [reply]
Patching is too cringe for me but thanks for the suggestion

**Comment by monholm** [comment]
Hi @bnussman 👋

You seem to have some experience using expo web, maybe you are able to track down the change that made it break? :) Once we narrow down what change made it break, we can start working on a PR :)

**Reply by YoussefHenna** [reply]
If you take a look at the steps [here](https://github.com/react-native-maps/react-native-maps/issues/4641#issuecomment-1473775067). You can see it's fixable by only importing the types into web and not the actual modules (i.e. an empty `index.web.js`, but a full `index.web.ts`). Maybe that could be possible by having a modified `index.web.ts` in the project's root?

**Comment by YoussefHenna** [comment]
Also worth mentioning [@teovilla/react-native-web-maps](https://github.com/teovillanueva/react-native-web-maps#teovillareact-native-web-maps) which is aimed to add web support for this library. You just alias 'react-native-maps' in the webpack config.

**Reply by lortschi** [reply]
The @teovilla/react-native-web-maps won't work with the common app.json metro bundler. It needs to split the project to native and regular react webpack structure.

**Comment by michjea** [comment]
Hi, I saw the postinstall script to solve the problem on issue #4641, but I am not sure how to use it. This is the first time I use React Native and Expo, so I am a big newbie. Is it possible to explain in more detail how to use this post-installation script?

**Comment by brada1703** [comment]
Has anyone figured out a work around?

**Reply by WatchmansEase** [reply]
https://medium.com/@pallavi8khedle/google-map-integration-with-react-native-expo-mobile-and-web-348a920c84c0

**Comment by SonGoki** [comment]
Please support MapView in web

**Comment by ydrea** [comment]
Here is my solution:
https://github.com/ydrea/eko

**Comment by boska** [comment]
Does anyone has a working solution? Plz

**Reply by rinaldycodes** [reply]
i think nothing

**Comment by david-fiaty** [comment]
+1

**Comment by salah-ghanim** [comment]
1. I'm happy to work on this next if there is enough requests for it.
2. please consider sponsoring this project if you want things done faster, with the current setup it's rather unrealistic to start such a big project after finishing the Fabric migration.

**Reply by xela6134** [reply]
Yes please

**Reply by yassinemeft** [reply]
We really need a solution, please!

**Reply by frozar** [reply]
We need to build an application which would be available on mobiles and web site and this feature is particularly important for us. Have you plan to work on this fix in near futur?

**Reply by antoinebrtd** [reply]
Yes please! Stuck on this, alternatives are not as good @salah-ghanim @GETAWAY-Dev

**Reply by salah-ghanim** [reply]
@antoinebrtd and everyone else, we're currently mid migration to support Fabric, my hope is that this will be done by end of the year and then hopefully this can be priortized.

I don't see the point of supporting new features when the current migration is not done and we're dealing with regressions because of it.

**Reply by KaleBBuR** [reply]
I would really appreciate this!

**Comment by ydrea** [comment]
I managed to solve it! 

https://gist.github.com/ydrea/27f70d9318981730be93948e2a54cb63

**Comment by vladimirmazarakis** [comment]
Really need web support +1

**Comment by federicogomez-estudiocactus** [comment]
We have a multi platform app with web used by a couple thousand people and web support is really, really important for us. We have been using `@teovilla/react-native-web-maps` without issues but it has left us stuck using webpack as our bundler which is now deprecated. It'd be great to have it

**Reply by ydrea** [reply]
https://github.com/ydrea/expo-leaflet
comments, reviews, pull requests and tests are very welcome!

**Comment by tiavina-mika** [comment]
Is there any news on this?
